mobilecoind/src/database.rs

```diff
@@ -223,6 +239,30 @@ impl Database {
         };
         Ok(())
     }
+
+    /// Get processed block information for a given (monitor id, block number).
+    pub fn get_processed_block(
+        &self,
+        monitor_id: &MonitorId,
+        block_num: u64,
+    ) -> Result<Vec<ProcessedTxOut>, Error> {
+        let db_txn = self.env.begin_ro_txn()?;
+
+        // Get monitor data to see if the monitor has synced this block.
+        let monitor_data = self.monitor_store.get_data(&db_txn, monitor_id)?;
+        if block_num < monitor_data.first_block {
+            return Err(Error::BlockIdTooSmall(block_num, monitor_data.first_block));
```

Comment:
maybe `BlockNumTooSmall` or BlockIndexTooSmall`? I don't think we are using `id` for this...